### PR TITLE
ImageTool: Fix ELF section loading and PIE relocations

### PR DESCRIPTION
**Pull request by mhaeuser**

### CODE REVIEW

File: BaseTools/ImageTool/ElfScan.c [hunk 1]
```diff
@@ -540,11 +533,16 @@ CreateIntermediate (
         return RETURN_VOLUME_CORRUPTED;
       }
 
+    if (!ProcessRelocSection (Shdr)) {
+      continue;
+    }
```
Comment (MikhailKrichanov) [inline BaseTools/ImageTool/ElfScan.c]: These 3 strings need 2 more spaces for proper alignment.

File: BaseTools/Conf/tools_def.template
Comment (MikhailKrichanov) [inline BaseTools/Conf/tools_def.template:1]: To make the same (regarding `-q` option) modification for AARCH64 you need to remove `-Wl,--emit-relocs` from `GCC_ARM_AARCH64_DLINK_COMMON` and add it to `GCC_ARM_DLINK_FLAGS`.

Comment (mhaeuser) [reply BaseTools/Conf/tools_def.template:1]: No - AARCH64 clearly is non-PIE by default:

https://github.com/acidanthera/audk/blob/b498c75ea8fd0b56e9cef87476f996f416ae17fc/BaseTools/Conf/tools_def.template#L724

https://github.com/acidanthera/audk/blob/b498c75ea8fd0b56e9cef87476f996f416ae17fc/BaseTools/Conf/tools_def.template#L726

File: BaseTools/ImageTool/ElfScan.c (hunk 1 above)
Comment (mhaeuser) [reply BaseTools/ImageTool/ElfScan.c]: Fixed, tyvm

File: BaseTools/Conf/tools_def.template
Comment (mhaeuser) [reply BaseTools/Conf/tools_def.template:1]: But why would I do that? This PR fixes the handling of the binaries as-is, as they have been tested for weeks. If we want to change AARCH64 (and possibly ARM) to PIE, this needs to be done as a separate change with separate testing and its own motivation/rationale. One step at a time.

**Pull request merged**